src/compile/mod.rs

```diff
@@ -147,71 +149,21 @@ impl fmt::Display for SolcVersion {
 pub struct Solc {
     /// Path to the `solc` executable
     pub solc: PathBuf,
-    /// The base path to set when invoking solc, see also <https://docs.soliditylang.org/en/v0.8.11/path-resolution.html#base-path-and-include-paths>
-    pub base_path: Option<PathBuf>,
-    /// Additional arguments passed to the `solc` exectuable
-    pub args: Vec<String>,
-}
-
-impl Default for Solc {
-    fn default() -> Self {
-        if let Ok(solc) = std::env::var("SOLC_PATH") {
-            return Solc::new(solc);
-        }
-
-        if let Some(solc) = Solc::svm_global_version()
-            .and_then(|vers| Solc::find_svm_installed_version(vers.to_string()).ok())
-            .flatten()
-        {
-            return solc;
-        }
-
-        Solc::new(SOLC)
-    }
-}
-
-impl fmt::Display for Solc {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "{}", self.solc.display())?;
-        if !self.args.is_empty() {
-            write!(f, " {}", self.args.join(" "))?;
-        }
-        Ok(())
-    }
+    /// Compiler version.
+    pub version: Version,
 }
 
 impl Solc {
-    /// A new instance which points to `solc`
-    pub fn new(path: impl Into<PathBuf>) -> Self {
-        Solc { solc: path.into(), base_path: None, args: Vec::new() }
-    }
-
-    /// Sets solc's base path
-    ///
-    /// Ref: <https://docs.soliditylang.org/en/v0.8.11/path-resolution.html#base-path-and-include-paths>
-    pub fn with_base_path(mut self, base_path: impl Into<PathBuf>) -> Self {
-        self.base_path = Some(base_path.into());
-        self
-    }
-
-    /// Adds an argument to pass to the `solc` command.
-    #[must_use]
-    pub fn arg<T: Into<String>>(mut self, arg: T) -> Self {
-        self.args.push(arg.into());
-        self
+    /// A new instance which points to `solc`. Invokes `solc --version` to determine the version.
+    pub fn new(path: impl AsRef<Path>) -> Result<Self> {
```

Comment:
this makes sense, this guarantees that the solc binary is usable.

should mention that this throws an error if the binary does not exist